tests/tasks/setup_mock_wifi_wpa3.yml

```diff
@@ -0,0 +1,105 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Install packages required to set up mock wifi network
+  package:
+    name:
+      - NetworkManager
+    state: present
+
+- name: Ensure NetworkManager is running
+  service:
+    name: NetworkManager
+    state: started
+
+- name: Copy server certificates
+  copy:
+    src: "{{ item }}"
+    dest: "/etc/pki/tls/{{ item }}"
+  with_items:
+    - server.key
+    - dh.pem
+    - server.pem
+    - cacert.pem
+
+
+- name: compile hostapd from src code
+  shell: |
+    mkdir -p /etc/hostapd/
+    yum -y install wget
+    wget https://w1.fi/releases/hostapd-2.7.tar.gz --no-check-certificate
+    tar xzvf hostapd-2.7.tar.gz
+    cd hostapd-2.7/hostapd/
+    echo "CONFIG_SAE=y" >> defconfig
+    cp defconfig .config
+    yum -y install libnl3-devel
+    yum -y install openssl-devel
+    yum -y install gcc
+    make -j 2
+    cd /root/
+    wget https://w1.fi/releases/wpa_supplicant-2.7.tar.gz --no-check-certificate
+    tar xzvf wpa_supplicant-2.7.tar.gz
+    cd wpa_supplicant-2.7/wpa_supplicant/
+    echo "CONFIG_SAE=y" >> defconfig
+    cp defconfig .config
+    echo '#include "crypto/aes-omac1.c"' >> ../src/common/wpa_common.c
+    make -j 2
+
+
+
+- name: Create hostapd config
+  copy:
+    content: |
+      interface=wlan1
+      driver=nl80211
+      ctrl_interface=/var/run/hostapd
+      ctrl_interface_group=0
+      ssid=mock_wifi
+      country_code=EN
+      hw_mode=g
+      channel=7
+      auth_algs=3
+      wpa=3
+      ieee8021x=1
+      eapol_version=1
+      wpa_key_mgmt=SAE
+      wpa_passphrase=p@55w0rD
+      eap_reauth_period=3600
+      eap_server=1
+      use_pae_group_addr=1
+      eap_user_file=/etc/hostapd/hostapd.eap_user
+      ca_cert=/etc/pki/tls/cacert.pem
+      dh_file=/etc/pki/tls/dh.pem
+      server_cert=/etc/pki/tls/server.pem
+      private_key=/etc/pki/tls/server.key
+      private_key_passwd=test
+      logger_syslog=-1
+      logger_syslog_level=0    
+      rsn_pairwise=CCMP
+    dest: /etc/hostapd/wireless.conf
+
+- name: Create eap_user_file config
+  copy:
+    content: |
+      * TLS
+    dest: /etc/hostapd/hostapd.eap_user
+
+- name: Load mac80211_hwsim kernel module to mock a wifi network
+  shell: modprobe mac80211_hwsim && sleep 5
```

Comment:
```
$ modinfo mac80211_hwsim | grep radios
parm:           radios:Number of simulated radios (int)
```
since your test is using wlan0 and wlan1, I suggest to add `radio=2` to the arguments of modprobe. On a similar test I found useful to "unblock" wifi using rfkill`, just to ensure that the SUT is not in "airplane mode"